sources/api/early-boot-config/src/provider/aws.rs

```diff
@@ -85,7 +85,7 @@ impl PlatformDataProvider for AwsDataProvider {
     ) -> std::result::Result<Vec<SettingsJson>, Box<dyn std::error::Error>> {
         let mut output = Vec::new();
 
-        let mut client = ImdsClient::new().await.context(error::ImdsClient)?;
+        let mut client = ImdsClient::new().context(error::ImdsClient)?;
```

Comment:
Since the `new()` method can't really fail now, I suggested below that it return a `Self` rather than a `Result`.  Once we do that we can remove the `.context(...)` here! :)